### ISSUE TITLE
Rewards chance -really low, really hard to calculate.

### ISSUE THREAD

**Issue by stgram** [graywolf336/CasinoSlots#30]
Hi, you got a really awesome plugin there!
But, I like math. And this causes me some trouble... :/
When I create slots, I want to know exactly how often is the player going to win 300$ for example, or the grand prize, or just a stack of melons.
Your plugin relies on getting 3 in a row/diagonal.
If we focus on the middle row for a sec:
Lets say in the config we have 20 blocks obsidian, 30 sand and 50 stone.
When you roll, you pick a random number from 0 to 99, picking the block. So the chance for an obsidian in the first column is 20/100 = 20%. But, to score a  double obsidian, we need to square that. For a winning row, the chance is (20/100)^3=0.8%
Ok, this could be the jackpot, but what about the others? The stone gets to 12.5%, and the sand is in the middle.
All we are left with is somewhere about 8/10 attempts that win nothing. And if I add more items, like 5 or 6 different winning blocks, the rate of loss rises to 95% and more! So, that is like, play 20, win one. Pretty dull for the player, right? The jackpot I guess should be somewhere about 100 000, to look nice for the players?
(if you want, I will create you a google excel sheet, so you can see how exactly the calculations go.)
But, then again, you can win by 5 different rows here. So that makes 5x the winning rate? Not really, since you got that neat little line of code that omits duplicates, and this is where I got lost in my math. You see, some times you win, some times you lose, but you cant really control what and when? 
That actually represents a real slot, good job. But too bad that our minecraft players aren't drunk rich kids (maybe just kids) who are willing to spend a fortune there.
What I am proposing, is a new system for "calculating" wins.
In the config, instead of number of blocks, add a variable between 0 and 0,4. Lets say, I add one block with a chance of 20%, one block with 12%, another with 8%, and the big prize, maybe 2%? The rest 58% are going to be empty, e.g. not matching rows. 
Those are my thoughts for now, but I will try to come up with more ideas.
So far, what is your opinion?

**Comment by graywolf336** [comment]
Curious to hear if you're still using the plugin and have come up with anything better than you had.

**Comment by stgram** [comment]
I'm no longer playing minecraft for quite the time now, but despite the initial negative comment I see I have left here, I actually managed to use your plugin quite nicely.
As far as I remember, I had created a script to randomize the block count and then test every one of them to calculate the chances, from which I then selected the ones I wanted to use. Kind of brute-forcing the combinations, instead of trying to reverse engineer the concept. It worked flawlessly, as I also mentioned, combined with an excel spreadsheet to calculate the total winning chances. If I ever go back to minecraft I will totally use the plugin again, gambling with items, money and command gifts(integration with everything else on the server) is the most addictive thing ever and the players love it.
